testsuite/pytests/sli2py_stimulating/test_iaf_ps_psp_poisson_generator_accuracy.py

```diff
@@ -0,0 +1,147 @@
+at different computation step sizes only differ because of limited
+machine precision.  The difference between the analytical result and
+the simulation, however, is dictated by the number of tics per
+millisecond.
+
+Author:  May 2005, February 2008, March 2009; Diesmann
+References:
+ [1] Morrison A, Straube S, Plesser H E, & Diesmann M (2007) Exact Subthreshold
+     Integration with Continuous Spike Times in Discrete Time Neural Network
+     Simulations. Neural Computation 19:47--79
+SeeAlso: testsuite::test_iaf_ps_psp_accuracy, testsuite::test_iaf_ps_dc_accuracy
+"""
+
+
+import math
+from math import exp
+
+import nest
+import pytest
+
+DEBUG = False
+
+# Global parameters
+T = 6.0
+tau_syn = 0.3
+tau_m = 10.0
+C_m = 250.0
+weight = 65.0
+delay = 1.0
+
+neuron_params = {
+    "E_L": 0.0,
+    "V_m": 0.0,
+    "V_th": 1500.0,
+    "I_e": 0.0,
+    "tau_m": tau_m,
+    "tau_syn_ex": tau_syn,
+    "tau_syn_in": tau_syn,
+    "C_m": C_m,
+}
+
+
+def alpha_fn(t):
+    if t < 0.0:
+        return 0.0
+    prefactor = weight * math.e / (tau_syn * C_m)
+    t1 = (exp(-t / tau_m) - exp(-t / tau_syn)) / (1 / tau_syn - 1 / tau_m) ** 2
+    t2 = t * exp(-t / tau_syn) / (1 / tau_syn - 1 / tau_m)
+    return prefactor * (t1 - t2)
+
+
+def spiketrain_response(spiketrain):
+    response = 0.0
+    for sp in spiketrain:
+        t = T - delay - sp
+        response += alpha_fn(t)
+    return response
+
+
+def create_spiketrain():
+    nest.ResetKernel()
+    nest.set(tics_per_ms=2**10, resolution=1)
+
+    pg = nest.Create("poisson_generator_ps", {"rate": 16000})
+    sr = nest.Create("spike_recorder")
+
+    nest.Connect(pg, sr)
+    nest.Simulate(T)
+    return sr.get("events", "times")
+
+
+spiketrain = create_spiketrain()
+
+
+@pytest.mark.parametrize("h", range(-10, 1, 2))
+def test_poisson_spikes_different_stepsizes(h):
+    nest.ResetKernel()
+
+    nest.set(tics_per_ms=2**10, resolution=2**h)
+
+    sg = nest.Create("spike_generator", {"start": 0, "spike_times": spiketrain, "precise_times": True})
+
+    neuron = nest.Create("iaf_psc_alpha_ps", params=neuron_params)
+    sr = nest.Create("spike_recorder")
+
+    if DEBUG:
+        mm = nest.Create("multimeter", params={"record_from": ["V_m"], "interval": 2**h})
+        nest.Connect(mm, neuron)
+
+    nest.Connect(sg, neuron, syn_spec={"weight": weight, "delay": delay})
+
+    nest.Simulate(T)
+
+    reference_potential = spiketrain_response(spiketrain)
+    if DEBUG:
+        u = neuron.get("V_m")
+        nest.Simulate(1.0)  # to get V_m recording until time T
+        times = mm.get("events", "times")
+        V_m = mm.get("events", "V_m")
+        import matplotlib.pyplot as plt
+
+        plt.plot(times, V_m)
+        plt.scatter([T], [u], s=20.0)
+        plt.scatter([T], [reference_potential], s=20, marker="X")
+        plt.show()
+        neuron.set(V_m=u)  # reset to value before extra 1s simulation
+
+    assert neuron.get("V_m") == pytest.approx(reference_potential)
```

Comment:
The original test checks that membrane potentials are within 1e-12 mV between different resolutions and within 1e-4 mV of the reference solution. Can you check what differences you actually get and then set an absolute tolerance here that is as tight as possible?

Reply:
Fixed